### PR TITLE
:memo: plan of new module

### PR DESCRIPTION
**Pull request by GabbasovDinar**

### CODE REVIEW

File: qcloud_sms/models/ir_config_parameter.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    _inherit = 'ir.config_parameter'
+
+    @api.model
+    def get_qcloud_sms_object(self):
+        _logger.debug(
+            'Tencent Cloud SMS Credentials: app_id=%s, app_key=%s',
+            self.get_param('qcloudsms.app_id', ''),
+            '%s...' % self.get_param('qcloudsms.app_key', '')[:5]
+        )
+        return QcloudSms(self.get_param('qcloudsms.app_id', ''), self.get_param('qcloudsms.app_key', ''))
```
Comment (yelizariev) [inline qcloud_sms/models/ir_config_parameter.py]: sudo

File: qcloud_sms/models/qcloud_sms.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+                                    default=fields.Datetime.now)
+    message = fields.Text(string='Message', required=True)
+    state = fields.Selection(STATE_SELECTION, string='Status', readonly=True, default='draft',
+                             help='Status of the SMS message')
+    template_id = fields.Many2one('qcloud.sms.template', 'SMS Template')
+
+    def _phone_get_country(self, partner):
+        if 'country_id' in partner:
+            return partner.country_id
+        return self.env.user.company_id.country_id
```
Comment (yelizariev) [inline qcloud_sms/models/qcloud_sms.py]: Why there is `phone` in method name?

File: qcloud_sms/models/qcloud_sms.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    name = fields.Char(string='Name', help='The template name')
+
+    domestic_sms_template_ID = fields.Integer(
+        string='Domestic SMS Template ID',
+        help='SMS Template ID is the Tencent Cloud SMS template (the specific content of the SMS message to be sent).'
+    )
+    domestic_template_params = fields.Text(
+        string="Domestic Template parameters",
+        help="Parameters must be separated by commas. If the template has no parameters, leave it empty."
+    )
```
Comment (yelizariev) [inline qcloud_sms/models/qcloud_sms.py]: It's not clear how to use params and what is params

File: qcloud_sms/models/qcloud_sms.py (hunk 2 above)
Comment (GabbasovDinar) [reply qcloud_sms/models/qcloud_sms.py]: @yelizariev like this https://github.com/odoo/odoo/blob/11.0/addons/phone_validation/models/phone_validation_mixin.py#L11

File: qcloud_sms/models/qcloud_sms.py (hunk 3 above)
Comment (GabbasovDinar) [reply qcloud_sms/models/qcloud_sms.py]: @yelizariev I have only this description of params https://cloud.tencent.com/document/product/382/5976

Comment (GabbasovDinar) [reply qcloud_sms/models/qcloud_sms.py]: @yelizariev params - array of template values. For example:
params = ['12345', '5']
template = "Your login verification code is {1}, which is valid for {2} minutes."

result = "Your login verification code is 12345, which is valid for 5 minutes."

File: qcloud_sms/models/ir_config_parameter.py (hunk 1 above)
Comment (GabbasovDinar) [reply qcloud_sms/models/ir_config_parameter.py]: @yelizariev I use the sudo parameter before calling the function
https://github.com/it-projects-llc/addons-dev/blob/20c9be39e6a4752a6ca79bb9fabcfcaf1e00ad0c/qcloud_sms/models/qcloud_sms.py#L97

Comment (yelizariev) [reply qcloud_sms/models/ir_config_parameter.py]: ok

File: qcloud_sms/models/qcloud_sms.py (hunk 3 above)
Comment (yelizariev) [reply qcloud_sms/models/qcloud_sms.py]: ok, but this explanation should be somewhere in fields description or at the form

**Pull request merged**